### PR TITLE
chore: creates a new examples landing page

### PR DESCRIPTION
**Pull request by krpeacock**
[Preview of Examples Page](https://deploy-preview-632--dfinity-docs-preview.netlify.app/docs/examples/index.html)

also moves examples up to higher prominence in sidenav
SDK-230

**Overview**
Why do we need this feature? What are we trying to accomplish?

**Requirements**
What requirements are necessary to consider this problem solved?

**Considered Solutions**
What solutions were considered?

**Recommended Solution**
What solution are you recommending? Why?

**Considerations**
What impact will this change have on security, performance, users (e.g. breaking changes) or the team (e.g. maintenance costs)?

### CODE REVIEW

**Review by samuelburri — APPROVED** [review]
LGTM

File: modules/examples/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+=== Beginner
+
+- https://github.com/dfinity/examples/tree/master/rust/pub-sub[Pub Sub]
+
+[[rust-advanced]]
+=== Advanced
+- https://github.com/timohanke/icvault[IC Vault] - end-to-end encryption on the IC
+
+
+More Rust examples coming soon!
```
Comment (olaszakos) [inline modules/examples/pages/index.adoc:52]: Here are some more official examples: https://github.com/dfinity/cdk-rs/tree/main/examples

Can we list videos here? For instance
* [Community Conversations | Overview of Building a Dapp in Rust](https://www.youtube.com/watch?v=6wyIhzsFbKw)
* [Community Conversations | Best Practices for Canisters in Rust](https://www.youtube.com/watch?v=36L33S_DYHY)

File: modules/examples/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+:proglang: Motoko
+:IC: Internet Computer
+:company-id: DFINITY
+ifdef::env-github,env-browser[:outfilesuffix:.adoc]
+
+[[example-code-intro]]
+In addition to our tutorials section, here are some quick references! You can also check out our https://github.com/dfinity/examples[Examples Repo] or https://github.com/dfinity/awesome-dfinity[Awesome Dfinity], a curated set of open-source projects and examples from our community!
+
+[[motoko]]
+== Motoko Examples
```
Comment (olaszakos) [inline modules/examples/pages/index.adoc:13]: Plug Motoko Playground?

**Review by olaszakos — COMMENTED** [review]
Looks good, left some comments

File: modules/ROOT/nav.adoc [hunk 3]
```diff
@@ -10,6 +10,10 @@
 ** xref:quickstart:local-quickstart.adoc[Local Development]
 ** xref:quickstart:network-quickstart.adoc[Network Deployment]
 
+* xref:examples:index.adoc[Example Code]
+*** link:https://github.com/dfinity/examples[Dfinity Examples^]
+*** link:https://github.com/dfinity/awesome-dfinity[Open-source Community Projects^]
```
Comment (olaszakos) [inline modules/ROOT/nav.adoc:15]: This repository hasn't been updated in 4 months now. We could try and maintain a list of good quality _and recent_ open source projects ourselves.

**Review by olaszakos — APPROVED** [review]

**Pull request merged**